mux_test.go

```diff
@@ -583,6 +583,26 @@ func TestMuxWith(t *testing.T) {
 	}
 }
 
+func TestRouterFromMuxWith(t *testing.T) {
+	t.Parallel()
+
+	r := NewRouter()
+
+	with := r.With(func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			next.ServeHTTP(w, r)
+		})
+	})
+
+	with.Get("/with_middleware", func(w http.ResponseWriter, r *http.Request) {})
+
+	ts := httptest.NewServer(with)
```

Comment:
how come you're reading a new server from `with` here? this line should be: `ts := httptest.NewServer(r)`

Reply:
I found this because I accidentally used it as the handler for an `http.Server`. `with` is a `Router`, which implements `http.Handler`, so why not?

It works as expected when `*Mux.With` properly initializes the `*Mux` that's being returned as a `Router` from that method. If that's not intended, should the type returned from `*Mux.With` be something that doesn't implement `http.Handler`?